settings.gradle

```diff
@@ -1,3 +1,23 @@
+pluginManagement {
+    repositories {
+        mavenCentral()
+        google()
+    }
+    resolutionStrategy {
+        eachPlugin {
+            if (requested.id.id == "com.android.application") {
+                useModule("com.android.tools.build:gradle:${requested.version}")
+            }
+        }
+    }
+    resolutionStrategy {
+        eachPlugin {
+            if (requested.id.id == "com.android.library") {
+                useModule("com.android.tools.build:gradle:${requested.version}")
+            }
+        }
+    }
```

Comment:
If I remember well AGP 4.2.0 publishes all required metadata so this can be removed after the upgrade